### ISSUE TITLE
support read replicas config

### ISSUE THREAD

**Issue by steebchen** [steebchen/prisma-client-go#1241]

**Comment by nettrino** [comment]
@steebchen even without this being implemented yet, is it possible to have `NewClient` accept a URL and thus to have two clients active, one for the read replica and one for the write replica just by means of passing different URLs?

**Comment by steebchen** [comment]
I haven't tried it but it might work yeah @nettrino

**Comment by nettrino** [comment]
I _think_ `client := db.NewClient(db.WithDatasourceURL(url))` might not be properly overwriting the URL if we have a single schema with a single datasource, however I will need to double check and come back to this ticket

**Comment by steebchen** [comment]
What happens when you initiate both with WithDatasourceURL and not set an env var outside of your application? Maybe it's overwriting it.

**Comment by nettrino** [comment]
I don't think this is doable to not set an env var at all - if I use an empty URL I get
```
"msg":"Error connecting prisma","error":"spawn: query engine errored: Error in data model: Diagnostics { errors: [DatamodelError { span: Span { start: 1125, end: 1127, file_id: FileId(0) }, message: \"Error validating datasource `db`: You must provide a nonempty URL\" }], warnings: [] }"}```

**Comment by nettrino** [comment]
@steebchen did you manage to replicate the above by any chance?
